File: frontend/src/types/checkout.ts
import { Product } from "./product";


export interface CheckoutCreate {
  order_id: number;
  payment_method: string;
  address?: string;
  email?: string;
  no_hp?: string;
}

export interface CheckoutResponse extends CheckoutCreate {
 id: number;
  order_id: number;
  user_id: number;
  product_id: number;
  total_price: number;
  payment_method: string;
  status: string;
  created_at: string;
  address: string;
  email: string;
  no_hp: string;
  product: Product; 
}

File: backend/schemas/checkout.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from schemas.product import ProductOut
from typing import Optional
from pydantic import EmailStr

class CheckoutBase(BaseModel):
    order_id: int
    payment_method: str
    address: Optional[str] = None  
    email: Optional[str] = None  
    no_hp: Optional[str] = None  

class CheckoutCreate(CheckoutBase):
    pass

class CheckoutResponse(CheckoutBase):
    id: int
    user_id: int
    total_price: int
    status: str
    created_at: datetime
    address: Optional[str] = None
    email: Optional[EmailStr] = None
    no_hp: Optional[str] = None
    product: Optional[ProductOut] = None  

    model_config = ConfigDict(from_attributes=True)
